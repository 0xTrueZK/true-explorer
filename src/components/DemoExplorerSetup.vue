<template>
    <v-container>
        <Demo-Explorer-Ready-Modal ref="demoExplorerReadyModal" />
        <v-row>
            <v-col>
                <v-card outlined>
                    <v-card-title class="justify-center">
                        <h1 class="logo">Ethernal</h1>
                    </v-card-title>
                    <v-card-text>
                        <p>
                            Ethernal is an open source block explorer for private EVM-based chains.
                        </p>
                        <p>
                            This tool lets you generate a demo block explorer from a RPC url.
                            It will stay up for 24h before getting deleted with all its data.<br><br>
                            You will have the option to subscribe to a plan to keep it up and running longer.
                        </p>
                        <p>
                            With the full version you'll be able to:
                            <ul>
                                <li>Use your own domain names</li>
                                <li>Customize the logo, the colors, add links, change the font, etc...</li>
                                <li>Display your native token total supply</li>
                                <li>Update your RPC & reset your explorer when needed</li>
                            </ul>
                        </p>
                    </v-card-text>
                </v-card>
            </v-col>
        </v-row>
        <v-row>
            <v-col>
                <v-card outlined>
                    <v-card-title>Set Up My Explorer</v-card-title>
                    <v-card-text>
                        <v-alert v-if="errorMessage" dense text type="error">{{ errorMessage }}</v-alert>
                        <v-form @submit.prevent="submit" v-model="valid">
                            <v-row>
                                <v-col cols="3">
                                    <v-text-field
                                        :rules="[v => !!v || 'Name is required']"
                                        outlined v-model="name" label="Explorer Name" placeholder="My Explorer Name" class="mb-2" required></v-text-field>
                                </v-col>

                                <v-col cols="6">
                                    <v-text-field
                                        :rules="[
                                            v => this.isUrlValid(v) || 'RPC needs to be a valid URL',
                                            v => !!v || 'RPC server is required'
                                        ]"
                                        outlined v-model="rpcServer" label="RPC URL" placeholder="https://my.rpc.com:8545" class="mb-2" required></v-text-field>
                                </v-col>

                                <v-col cols="3">
                                    <v-text-field outlined v-model="nativeToken" label="Native Token Name (optional)" placeholder="ether" class="mb-2" required></v-text-field>
                                </v-col>
                            </v-row>
                            <v-btn large width="100%" :loading="loading" color="primary" :disabled="!valid" type="submit">Generate Explorer</v-btn>
                        </v-form>
                    </v-card-text>
                </v-card>
            </v-col>
        </v-row>
    </v-container>
</template>

<script>
import DemoExplorerReadyModal from './DemoExplorerReadyModal';

export default {
    name: 'DemoExplorerSetup',
    components: {
        DemoExplorerReadyModal
    },
    data: () => ({
        valid: false,
        name: null,
        rpcServer: null,
        loading: false,
        nativeToken: 'ether',
        errorMessage: null
    }),
    mounted() {
    },
    methods: {
        submit() {
            this.loading = true;
            this.errorMessage = null;
            this.server.createDemoExplorer(this.name, this.rpcServer, this.nativeToken)
                .then(({ data }) => {
                    this.$refs.demoExplorerReadyModal.open({ domain: data.domain });
                })
                .catch(error => {
                    console.log(error)
                    this.errorMessage = error.response && error.response.data || 'Error while creating explorer. Please retry.'
                })
                .finally(() => this.loading = false);
        },
        isUrlValid(url) {
            try {
                new URL(url);
                return true;
            } catch(error) {
                return false;
            }
        },
    },
    computed: {}
};
</script>
<style scoped lang="scss">
.v-application {
    background: #f7f7f7
}
</style>
