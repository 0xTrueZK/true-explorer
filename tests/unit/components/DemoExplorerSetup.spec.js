import MockHelper from '../MockHelper';

import DemoExplorerSetup from '@/components/DemoExplorerSetup.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('DemoExplorerSetup.vue', () => {
    it('Should display demo explorer setup page', async () => {
        const wrapper = helper.mountFn(DemoExplorerSetup);

        expect(wrapper.html()).toMatchSnapshot();
    });
});
