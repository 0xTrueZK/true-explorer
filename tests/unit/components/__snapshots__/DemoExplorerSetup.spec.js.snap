// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoExplorerSetup.vue Should display demo explorer setup page 1`] = `
<div class="container">
  <div class="v-dialog__container">
    <!---->
  </div>
  <div class="row">
    <div class="col">
      <div class="v-card v-sheet v-sheet--outlined theme--light">
        <div class="v-card__title justify-center">
          <h1 class="logo">Ethernal</h1>
        </div>
        <div class="v-card__text">
          <p>
            Ethernal is an open source block explorer for private EVM-based chains.
          </p>
          <p>
            This tool lets you generate a demo block explorer from a RPC url.
            It will stay up for 24h before getting deleted with all its data.<br><br>
            You will have the option to subscribe to a plan to keep it up and running longer.
          </p>
          <p>
            With the full version you'll be able to:
          <ul>
            <li>Use your own domain names</li>
            <li>Customize the logo, the colors, add links, change the font, etc...</li>
            <li>Display your native token total supply</li>
            <li>Update your RPC &amp; reset your explorer when needed</li>
          </ul>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col">
      <div class="v-card v-sheet v-sheet--outlined theme--light">
        <div class="v-card__title">Set Up My Explorer</div>
        <div class="v-card__text">
          <!---->
          <form novalidate="novalidate" class="v-form">
            <div class="row">
              <div class="col col-3">
                <div class="v-input mb-2 theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <fieldset aria-hidden="true">
                        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                      </fieldset>
                      <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;">Explorer Name</label><input required="required" id="input-9" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="col col-6">
                <div class="v-input mb-2 theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <fieldset aria-hidden="true">
                        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                      </fieldset>
                      <div class="v-text-field__slot"><label for="input-12" class="v-label theme--light" style="left: 0px; position: absolute;">RPC URL</label><input required="required" id="input-12" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div class="col col-3">
                <div class="v-input mb-2 v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <fieldset aria-hidden="true">
                        <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                      </fieldset>
                      <div class="v-text-field__slot"><label for="input-15" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Native Token Name (optional)</label><input required="required" id="input-15" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div> <button type="submit" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--large" style="width: 100%;"><span class="v-btn__content">Generate Explorer</span></button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
